fog/overseer/server/src/config.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+use mc_common::ResponderId;
+use mc_fog_sql_recovery_db::SqlRecoveryDbConnectionConfig;
+use mc_fog_uri::FogIngestUri;
+use serde::Serialize;
+use structopt::StructOpt;
+
+/// StructOpt configuration options for an Overseer Server
+#[derive(Clone, Serialize, StructOpt)]
+pub struct OverseerConfig {
+    /// Host to listen on.
+    #[structopt(long, default_value = "127.0.0.1")]
+    pub listen_host: String,
+
+    /// Port to start webserver on.
+    #[structopt(long, default_value = "9090")]
+    pub listen_port: u16,
+
+    /// Local Ingest Node ID
+    #[structopt(long)]
+    pub local_overseer_node_id: ResponderId,
+
+    /// TODO: Make this an environment variable that can be dynamically
```

Comment:
For context, this idea came up in a discussion with ops about how to keep one Fog Overseer instance running between blue-green deployments. 

I had, perhaps naively, thought that there'd be a widely-used, OOTB way to achieve this in Rust. After digging in a little more, I found that this isn't the case, but there is this crate, [spirit](https://docs.rs/spirit/0.3.7/spirit/), that appears to allow for config reloading. What do you think about this crate? Not sure if we've looked into this before? 

@jgreat Feel free to add anything re: our discussion on dynamic config reloading.

Reply:
Why not send configuration changes over an HTTP API?

Reply:
I tend to think that configuration shouldn't be modified over HTTP and that we should try to keep configs in a text file. That way, anyone can look at that text file at any given time to see what the configs are. However, I'd be happy to do HTTP given it's simplicity / if ops is ok with it.

Reply:
Text file is different than an env variable though. I view env variable as "dynamic" in the same way I'd consider HTTP dynamic, and a text file as "static"

Reply:
Could you not save the active configs to a text file even if it's loaded over an http api?

Reply:
Ah, I was using text file == env variable because most env variables are set in some sort of .env file. The way I envision it, updates to configs would be made by editing the text file (making it dynamic and not static). Then a worker would check for an updated config. 

You could also save the active configs to a text file, but at that point why not just have the person changing the config edit the text file directly?

Reply:
I'm curious as to what @jgreat thinks about this. Would you prefer to update the list of Fog Ingest nodes that Overseer watches through (a) HTTP request or (b) updating a config file and having Overseer check for these updates?

Reply:
Maybe it could do something with like, a shared volume i guess? That might be durable? Lets see what @jgreat says

Reply:
@joekottke this is the comment I mentioned offline.

Reply:
@garbageslam @samdealy We can very easily map in a config into the running environment.  It will also follow a given running instance around even if k8s moves the running instance to another machine.  The file effectively gets mapped in as an external volume.  Another nice effect of this is that we can _change_ the running config file, k8s will update that file on the running instance(s) (after a few seconds as changes get pushed out).  The on-disk semantics of this are handled through symlinks:  The running config is mapped via a symlink to a file, and then the new file gets mapped in, and the symlink gets updated.  One caveat: if the 'config file watcher' is only watching a specifc on-disk inode (after following the symlink reference) then it _might not_ notice the updated file (because it is at a new inode).  Just something to keep in mind.

Reply:
Some other things to keep in mind:
* k8s will have to reference the same config file contents for both the active and inactive nodes, at least the way we have things deployed now.
* The config file is read-only